examples/bert/bert_train.py

```diff
@@ -371,11 +400,76 @@ def decode_record(record):
     return example
 
 
+def get_checkpoint_callback():
+    if FLAGS.checkpoint_save_directory is not None:
+        if FLAGS.use_cloud_storage:
+            storage_client = storage.Client()
+            bucket = storage_client.get_bucket(FLAGS.gcs_bucket)
+            blobs = bucket.list_blobs(prefix=FLAGS.checkpoint_save_directory)
+            if FLAGS.skip_restore:
+                for blob in blobs:
+                    blob.delete()
+            checkpoint_path = (
+                "gs://"
+                + FLAGS.gcs_bucket
+                + "/"
+                + FLAGS.checkpoint_save_directory
+            )
+            return tf.keras.callbacks.BackupAndRestore(
+                backup_dir=checkpoint_path,
+            )
+
+        else:
+            if os.path.exists(FLAGS.checkpoint_save_directory):
+                if not os.path.isdir(FLAGS.checkpoint_save_directory):
+                    raise ValueError(
+                        "`checkpoint_save_directory` should be a directory, "
+                        f"but {FLAGS.checkpoint_save_directory} is not a "
+                        "directory. Please set `checkpoint_save_directory` as "
+                        "a directory."
+                    )
+
+                elif FLAGS.skip_restore:
+                    # Clear up the directory if users want to skip restoring.
+                    shutil.rmtree(FLAGS.checkpoint_save_directory)
+            checkpoint_path = FLAGS.checkpoint_save_directory
+            return tf.keras.callbacks.BackupAndRestore(
+                backup_dir=checkpoint_path,
+            )
+
+
+def get_tensorboard_callback():
+    timestamp = datetime.datetime.now().strftime("%Y%m%d-%H%M%S")
+    if FLAGS.use_cloud_storage:
+        log_dir = (
+            "gs://"
+            + FLAGS.gcs_bucket
+            + "/"
+            + FLAGS.tensorboard_log_path
+            + timestamp
+        )
+    else:
+        log_dir = FLAGS.tensorboard_log_path + timestamp
+    return tf.keras.callbacks.TensorBoard(log_dir=log_dir, histogram_freq=1)
+
+
 def main(_):
-    print(f"Reading input data from {FLAGS.input_files}")
-    input_filenames = list_filenames_for_arg(FLAGS.input_files)
+    if FLAGS.use_cloud_storage:
```

Comment:
Can we autodetect this? Or use a logging API that will route things the right way?

I remember @haifeng-jin was looking at logging a bit ago. Do you know of any good solution for cloud logging?